File: services/gateways/bff/src/modules/interview/usecases/resume-interview.usecase.ts
import { Injectable } from "@nestjs/common";
import { InterviewGrpcService } from "../../../infra/grpc/services/interview-grpc.service";

export class ResumeInterviewCommand {
    constructor(public readonly interviewId: string) {}
}

@Injectable()
export class ResumeInterviewUseCase {
    constructor(private readonly interviewGrpcService: InterviewGrpcService) {}

    async execute(command: ResumeInterviewCommand): Promise<{
        interviewId: string;
        status: string;
        resumedAt: string;
    }> {
        return await this.interviewGrpcService.resumeInterview(command.interviewId);
    }
}

File: services/domains/interview/src/main/java/me/unbrdn/core/interview/application/port/in/ResumeInterviewUseCase.java
package me.unbrdn.core.interview.application.port.in;

import java.util.UUID;

/** Use Case for resuming an interview */
public interface ResumeInterviewUseCase {
    ResumeInterviewResult execute(ResumeInterviewCommand command);

    record ResumeInterviewCommand(UUID interviewId) {}

    record ResumeInterviewResult(String interviewId, String status, String resumedAt) {}
}
